native/cocos/core/scene-graph/Node.h

```diff
@@ -63,19 +63,14 @@ class Node : public BaseNode {
     public:
         ~UserData() override = default;
 
-    protected:
```

Comment:
Why remove protected ?
We keep UserData's constructor protected since we don't wanna developer uses `new UserData()`. Instead, developer should inherit UserData. `new UserData()` is meaningless

Reply:
I will revert it. Remove it to do debug.